src/libraries/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceInternal.cs

```diff
@@ -6,21 +6,24 @@
 using System.IO;
 using System.Collections;
 using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
 
 namespace System.Diagnostics
 {
     internal static class TraceInternal
     {
         private class TraceProvider : DebugProvider
         {
-            public override void Fail(string message, string detailMessage) { TraceInternal.Fail(message, detailMessage); }
+#pragma warning disable CS8770 // Method lacks `[DoesNotReturn]` annotation to match overridden member.
+            public override void Fail(string? message, string? detailMessage) { TraceInternal.Fail(message, detailMessage); }
+#pragma warning restore CS8770
             public override void OnIndentLevelChanged(int indentLevel)
             {
                 lock (TraceInternal.critSec)
                 {
-                    foreach (TraceListener listener in Listeners)
+                    foreach (TraceListener? listener in Listeners)
                     {
-                        listener.IndentLevel = indentLevel;
+                        listener!.IndentLevel = indentLevel;
```

Comment:
The list doesn't allow null, so iteration wouldn't produce null, anyways was need to do it in so many `foreach` loop, hope there were a way to say that loop iteration wouldn't return null ...

Reply:
Perhaps add a reference to the tracking issue? https://github.com/dotnet/csharplang/issues/3214

Reply:
Sure